### PR TITLE
Initial MSQ Templates

### PR DESCRIPTION
**Pull request by OutKa5t**
Added template file for MSQL
Added test and parameter file for MSQL
Added initial reeadme
Added NSG template for MSQL

Ticket NCSD-1029

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# dfc-shared
```
Comment (rivoric) [inline README.md]: It's not dfc-shared

File: README.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# dfc-shared
+
+[![Build status](https://sfa-gov-uk.visualstudio.com/Digital%20First%20Careers/_apis/build/status/DFC%20Shared/dfc-shared?branchName=master)](https://sfa-gov-uk.visualstudio.com/Digital%20First%20Careers/_build/latest?definitionId=1690)
```
Comment (rivoric) [inline README.md]: Link to dfc-shared build

**Review by rivoric — COMMENTED** [review]
Readme still referencing dfc-shared

File: template.json [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    "managedSQLNSG": "[concat(parameters('sharedPrefix'),'-nsg')]",
+    "msqlSubnetName": "[concat(parameters('sharedPrefix'),'-sn')]",
+    "msqlVnetName": "[concat(parameters('sharedPrefix'),'-vnet')]"
+  },
+  "resources": [
+    {
+      "name": "sharedManagedInstanceEnvironment",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2017-05-10",
+      "condition": "[equals(parameters('envType'),'dev')]",
```
Comment (rivoric) [inline template.json]: I thought you needed this deploying to preprod? If so you can remove the envType parameter completely

File: template.json [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          "virtualNetworkNonRoutedSubnets": {
+            "value": "[parameters('virtualNetworkNonRoutedSubnets')]"
+          }
+        }
+      }
+    },
+    {
+      "name": "managedSQLNSG",
+      "type": "Microsoft.Resources/deployments",
+      "condition": "[parameters('deployManageSql')]",
```
Comment (rivoric) [inline template.json]: Is there ever a case where you are going to run this and not want to deploy the managed SQL? Assuming not then you can completely remove the deployManageSql param

File: parameters.json [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    "msqlSubnetName": {
+      "value": "__msqlSubnetName__"
+    },
+    "msqlNetworkSecurityGroup": {
+      "value": "__msqlNetworkSecurityGroup__"
+    },
+    "deployManageSql": {
+      "value": "__deployManageSql__"
+    },
+    "msqlResourceGroupName": {
```
Comment (rivoric) [inline parameters.json]: msqlResourceGroupName is not used anywhere

File: test-parameters.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    "msqlSubnetName": {
+      "value": "dfc-local-msql"
+    },
+    "msqlNetworkSecurityGroup": {
+      "value": "dfc-local-msql-nsg"
+    },
+    "deployManageSql": {
+      "value": true
+    },
+    "msqlResourceGroupName": {
```
Comment (rivoric) [inline test-parameters.json]: msqlResourceGroupName is not used anywhere

File: test-parameters.json [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    "threatDetectionEmailAddress": {
+      "value": []
+    },
+    "msqlVnetName": {
+      "value": ""
+    },
+    "msqlSubnetName": {
+      "value": "dfc-local-msql"
+    },
+    "msqlNetworkSecurityGroup": {
```
Comment (rivoric) [inline test-parameters.json]: msqlNetworkSecurityGroup is not used anywhere

File: parameters.json [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    "threatDetectionEmailAddress": {
+      "value": "__threatDetectionEmailAddress__"
+    },
+    "msqlVnetName": {
+      "value": "__msqlVnetName__"
+    },
+    "msqlSubnetName": {
+      "value": "__msqlSubnetName__"
+    },
+    "msqlNetworkSecurityGroup": {
```
Comment (rivoric) [inline parameters.json]: msqlNetworkSecurityGroup

File: parameters.json [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    "sqlServerActiveDirectoryAdminObjectId": {
+      "value": "__sqlAdGroupObjectID__"
+    },
+    "virtualNetworkRoutedSubnets": {
+      "value": "__virtualNetworkRoutedSubnets__"
+    },
+    "virtualNetworkNonRoutedSubnets": {
+      "value": "__virtualNetworkNonRoutedSubnets__"
+    },
+    "threatDetectionEmailAddress": {
```
Comment (rivoric) [inline parameters.json]: threatDetectionEmailAddress is not used anywhere

File: test-parameters.json [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+      "value": [
+        "dfc-local-msql"
+      ]
+    },
+    "virtualNetworkNonRoutedSubnets": {
+      "value": [
+        "dfc-local-nrs"
+      ]
+    },
+    "threatDetectionEmailAddress": {
```
Comment (rivoric) [inline test-parameters.json]: threatDetectionEmailAddress is not used anywhere

File: template.json (hunk 4 above)
Comment (OutKa5t) [reply template.json]: Maybe, if the network needs to change.

File: parameters.json (hunk 5 above)
Comment (OutKa5t) [reply parameters.json]: Dropped

File: template.json (hunk 3 above)
Comment (OutKa5t) [reply template.json]: Dropped

File: README.md (hunk 1 above)
Comment (OutKa5t) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (OutKa5t) [reply README.md]: Fixed.

**Review by rivoric — APPROVED** [review]

File: test-parameters.json (hunk 10 above)
Comment (OutKa5t) [reply test-parameters.json]: Removed

**Pull request merged**